napari/layers/shapes/shapes.py

```diff
@@ -297,6 +324,43 @@ class Shapes(Layer):
     # in the thumbnail
     _max_shapes_thumbnail = 100
 
+    _drag_modes = {
+        Mode.SELECT: select,
+        Mode.DIRECT: select,
+        Mode.VERTEX_INSERT: vertex_insert,
+        Mode.VERTEX_REMOVE: vertex_remove,
+        Mode.ADD_RECTANGLE: add_rectangle,
+        Mode.ADD_ELLIPSE: add_ellipse,
+        Mode.ADD_LINE: add_line,
+        Mode.ADD_PATH: add_path_polygon,
+        Mode.ADD_POLYGON: add_path_polygon,
```

Comment:
I made all these privates as we could also have multiple callback per mode, but here it was not the case, 
so I use just callables as values.

Reply:
private is good for me